### ISSUE TITLE
[Bug] Bold Text Accessibility option breaks Text Renders

### ISSUE THREAD

**Issue by CarstenB-ai** [Vali-98/ChatterUI#161]
Using Android 13, and the letters of TextViews are being cut off at the end of them a lot of the times. I've attached a screenshot for reference.
![Screenshot_20241229-113717_Trebuchet](https://github.com/user-attachments/assets/90a634bc-c1bc-486f-87c2-5af550134cfa)
Oh yeah, the version number is also cut off, but it is v0.8.3a, so the latest one, but I tried every single version back to v0.8.0 and it happens on all of them, so quite wondering why nobody else has pointed this out...

**Comment by Vali-98** [comment]
This is the first time I've seen an issue like this. Is your default OS language english? And does this occur on 0.7.10?

**Comment by CarstenB-ai** [comment]
> This is the first time I've seen an issue like this. Is your default OS language english? And does this occur on 0.7.10?

Hello there! Yes, my Operating Systems language is set to "English (United States)". I've tried  different font-sizes and UI-sizes in the OS-Settings, but that's also not it. I did just now try out the v0.7.10, and yes, the issue persists. Though, I've noticed differences, like in the v0.8.X, the username is cut off, and responses are missing a huge chunk of text (Like about 5% of the responses end is cut off). In v0.7.X, the responses seem to be complete, and the usernames "User" as well, but can't try with a longer name, because usernames weren't changable in v0.7.X apparently.

I hope this helps somehow, never seen something like this myself. Attached is another screenshot of v0.7.10:
![Screenshot_20241230-092934_Trebuchet](https://github.com/user-attachments/assets/55efe47b-2870-415c-9bf4-37f37faf6b9b)

**Comment by Vali-98** [comment]
I honestly have no idea why this occurs. Are you using some custom ROM for your device? Your font scaling and font weight do not look like any common vendor OS.

**Comment by CarstenB-ai** [comment]
Nah, just a Samsung with Android 13. You can change font scaling, display size, and fonts itself, but the problem also applies on default settings sadly. If I show layout bounds, it doesn't tell much either sadly. Maybe giving the TextViews in the Drawer a weight of 1 (I see that you are most likely using wrap_content for these) would fix at least those texts. I'd be willing to test out such a build if you want to, but I'm not sure if that's the problem, nor how to fix it for all the other texts.

**Comment by Vali-98** [comment]
Apparently this is an ancient react-native bug:

https://github.com/facebook/react-native/issues/15114

There are a few possible solutions, but this is most probably best delt with by updating the app to used a stricter font and unify styling across text fields.

**Comment by genewitch** [comment]
can confirm on samsung galaxy note 20 (non-ultra):

![Image](https://github.com/user-attachments/assets/ac06a264-aab6-4b7b-a6bb-d82faf966451)

not sure if related but the end of generated paragraphs is sometimes cut off as well, whereas it is not in the service software (i use LM Studio). Since both involve truncating the end of things, maybe there's something in common?

**Comment by paboum** [comment]
Also if "friendly name" of API entry is too long, the "edit" button goes outside the screen and can't be tapped.

**Comment by Vali-98** [comment]
Hey there, just wanted to check in on this issue post 0.8.5, still a problem?

**Comment by CarstenB-ai** [comment]
![Image](https://github.com/user-attachments/assets/0409585c-c587-40a5-91be-a5c8b3ef49a7)
![Image](https://github.com/user-attachments/assets/ae903bd5-dfac-4c59-a93f-44b836e7fa1b)

Believe me how much I'd love to tell ya it's fixed, sadly though, it's not. I also included a settings screenshot, since I've noticed no letters being cut off, but it looks like sometimes whole words are cut instead.

**Comment by Vali-98** [comment]
Thanks for the follow up, as saddening as it is!

Its really weird, since some words are getting cut off like `TTS`, despite longer words like `Model` being just fine.

Perhaps the next step is to see if the localization rework will allow me to wrap all text in a themed text context to set a default font for everything.

**Comment by Vali-98** [comment]
One more thing, is the `bold text` accessibility option enabled here? I've always noticed the text in the screenshots being much more bold than normal.

**Comment by CarstenB-ai** [comment]
Well, indeed... Smh turning off bold text fixes the issue. Now, that's weird. Anyways, so we know bold text is the issue, good job! Maybe you can fix it to be used with that setting smh, I believe in you!

**Comment by Vali-98** [comment]
This should finally be fixed by 97f7a41333439c6b087cfce4305eaa02e4667a0c

For now, it forces the app to use non-bold text. Ideally the app would support both bold accessibility, but this is a decent temporary solution.
